File: jpa-fetch-join/src/main/java/com/example/jpafetchjoin/coupon/Coupon.java
package com.example.jpafetchjoin.coupon;


import com.example.jpafetchjoin.order.Order;
import com.fasterxml.jackson.annotation.JsonIgnore;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "coupon")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class Coupon {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private long id;

  @Column(name = "discount", nullable = false)
  private double discount;

  @ManyToOne
  @JoinColumn(name = "order_id", nullable = false)
  @JsonIgnore
  private Order order;


}

File: event-transaction/src/main/kotlin/com/example/eventtransaction/coupon/Coupon.kt
package com.example.eventtransaction.coupon

import com.example.eventtransaction.EntityAuditing
import java.math.BigDecimal
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table

@Entity
@Table(name = "coupon")
class Coupon(
    @Column(name = "amount", nullable = false)
    var amount: BigDecimal
) : EntityAuditing() {

}